web/src/components/molecules/Settings/Project/Dataset/DatasetItem/index.tsx

```diff
@@ -13,17 +13,30 @@ export type Props = {
   id: string;
   name: string;
   removeDatasetSchema?: (schemaId: string) => void;
+  onDownloadFile?: (id: string, name: string) => void;
 };
 
-const DatasetItem: React.FC<Props> = ({ className, id, name, removeDatasetSchema }) => {
+const DatasetItem: React.FC<Props> = ({
+  className,
+  id,
+  name,
+  removeDatasetSchema,
+  onDownloadFile,
+}) => {
   const t = useT();
-
   const [isHover, setHover] = useState(false);
   const [isVisible, setIsVisible] = useState(false);
+  const [isDownloading, setDownloading] = useState(false);
+
   const handleRemoveDatasetSchema = useCallback(
     () => removeDatasetSchema?.(id),
     [id, removeDatasetSchema],
   );
+  const handleDownloadFile = useCallback(() => {
+    setDownloading(true);
+    onDownloadFile?.(id, name);
```

Comment:
I think `onDownloadFile` is async function. If so, it have to wait until the function is completed by using `await`.

Reply:
'await' has no effect on the type of this expression because the onDownloadFile function is not returning a promise. so I handled async/await in onDownloadFile itself , but I added a callback here to maintain a spinning effect , would you please check